src/polyhedralGravity/util/UtilityContainer.h

```diff
@@ -309,6 +309,24 @@ namespace polyhedralGravity::util {
         return 0.5 * euclideanNorm(cross(triangle[1] - triangle[0], triangle[2] - triangle[0]));
     }
 
+    /**
+     * Calculates the magnitude between two values and return true if the magnitude
+     * between the exponents in greater than 17.
+     * @tparam T - numerical type
+     * @param first - first number
+     * @param second - second number
+     * @return true if the difference is too be huge, so that floating point absorption will happen
+     */
+    template<typename T>
+    bool isCriticalDifference(const T &first, const T &second) {
+        // 57 is the (log2) exponent of the floating point (1 / 1e-17)
+        constexpr int maxExponentDifference = 57;
```

Comment:
17 seems a bit ambitious? shouldn't it be 15 or 16? https://stackoverflow.com/questions/9999221/double-precision-decimal-places

Reply:
GitHub did not save my response, so writing it once more again...


@gomezzz Agreed.

In the beginning, I had 14 as the allowed exponent difference, but that was too strict. I relaxed it now from 17 to 15.
This should make the whole thing work fine and not print any false positive warning as it does with 14 when, e.g., executing the cube plotting script.

### Rounding errors introduced by Python --> C++

Thanks to this circumstance and a bit of trying, I also found an unknown source of inaccuracy until now. This only happens when using the Python interface rather than when directly using the C++ implementation.

When the numbers are handed over from C++ to Python via `pybind11` the floating points will be affected by rounding errors:

Actual Example:

Point in Python: [1.0, 2.0, 0.0]
Point in C++: [1.0000000000000026, 2.0, 0.0]


This could be a problem when the point falls, e.g., on the edge of the body, but it does not anymore due to rounding errors. This could be a severe issue since the application of the singularity terms depends on this fact.

Rounding errors were the reason we introduced the EPSILON of 1e-14 to C++ implementation. This effectively counters any issues related to that. This EPSILON perfectly fits in the case above, but if the 15th digit were shifted to the left, there would be a problem!

I suggest not doing anything since these edgy* (geometrically spoken) cases are rare, and the current EPSILON of 1e-14 does an excellent job. Especially since increasing the EPSILON would not necessarily lead to better results since then values that should be distinct become similar... and the issue would arise again, but only in reverse...

So, I argue we're in the *golden mid* of trade-offs here :) But we can keep this in mind for whatever edge case the future may bring.